crates/gitbutler-tauri/src/open.rs

```diff
@@ -0,0 +1,66 @@
+use crate::error::Error;
+use anyhow::anyhow;
+use regex::Regex;
+use std::env;
+use tracing::instrument;
+
+pub fn open_that(path: &str) -> Result<(), Error> {
+    let re = Regex::new(r"^((https://)|(http://)|(mailto:)|(vscode://)|(vscodium://)).+").unwrap();
+    if !re.is_match(path) {
+        return Err(anyhow!("Invalid path format").into());
+    }
+
+    fn clean_env_vars(var_names: &[&str]) {
+        for var_name in var_names {
+            if let Ok(var_value) = env::var(var_name) {
+                let cleaned_value = var_value
+                    .split(':')
+                    .filter(|path| !path.contains("appimage-run"))
+                    .collect::<Vec<_>>()
+                    .join(":");
+                env::set_var(var_name, cleaned_value);
```

Comment:
The environment of the running process should not be affected - it's an unexpected side-effect.
It's certainly OK to read, but the new values should be written to the 'about-to-be-started' process only.

Reply:
Hmm okay yeah I get that this is not a good programming practice, but this is also the crux of this bug fix. See the appimage is basically a zip file that packages in all the code and libraries required to run gitbutler. Including all the `lib*.so` C libraries. 

To make those work, it also prepends a bunch of paths to important env vars like `LD_LIBRARY_PATH` and `XDG_DATA_DIRS`. So that, for example, `XDG_DATA_DIRS` is no longer `XDG_DATA_DIRS=/usr/share/gnome:/usr/local/share:/usr/share:/var/lib/snapd/desktop`, but inside the appimage is then `XDG_DATA_DIRS=/run/appimage-run/abc123/usr/lib:/usr/share/gnome:/usr/local/share:/usr/share:/var/lib/snapd/desktop`.

So in the case of `xdg-open`, it executes that bash script which is on the host system, with the env vars modified by the appimage, and then subprocesses of `xdg-open` are looking for their libraries in incorrect places. 

In my local system's case specifically, `xdg-open` calls `gdbus` to open a lib and `gdbus` finds one of its dynamically linked libraries in the `../appimage-run/*` path instead of the host's normal library path, there's a version mismatch, and it fails due to not being able to find some function. I ran into very similar errors in Ubuntu and Fedora. [This guy](https://github.com/vrc-get/vrc-get/pull/1260#issuecomment-2242153215) has the same thing with openssl when `xdg-open` uses `kde-open` on KDE

Reply:
So long story short, I guess we could place these `appimage-run` paths at the end of the env var's, but either way, they need to be modified to fix this bug :/

Reply:
Ahh wait sorry I just reread your comment - yes, the idea is to only modify it for the process it's about to start haha. I thought that was the case here since we're modifying the `cmd::env`, but I see where I messed up with this filter fn. 

I'll give this another look over this weekend 👌

Reply:
Okie doke, I think I've got it squared away finally :pray: